cra-rxjs-styled-components/src/components/repo-meta/RepoMeta.tsx

```diff
@@ -0,0 +1,63 @@
+import { formatDistance } from 'date-fns';
+import { getColourForLanguage } from '../../helpers/colours';
+import { BranchLogo } from '../misc/BranchLogo';
+import { StarLogo } from '../misc/StarLogo';
+import {
+  LanguageColor,
+  Metadata,
+  SocialCount,
+  TextSpan,
+  SocialWrapper,
+} from './RepoMeta.styles';
+interface RepoMetaProps {
+  language?: string | null;
+  stargazerCount: number;
+  forkCount: number;
+  updatedAt: Date;
+}
+
+function RepoMeta({
+  language,
+  stargazerCount,
+  forkCount,
+  updatedAt,
+}: RepoMetaProps) {
+  return (
+    <Metadata>
+      {language && (
+        <div>
+          <LanguageColor
+            style={{
+              backgroundColor: getColourForLanguage(language) || '#ccc',
+            }}
+          />
+          {language}
+        </div>
+      )}
+      {(stargazerCount > 0 || forkCount > 0) && (
+        <SocialWrapper>
+          {stargazerCount > 0 && (
+            <SocialCount>
+              <StarLogo />
+              <TextSpan>{stargazerCount}</TextSpan>
+            </SocialCount>
+          )}
+          {forkCount > 0 && (
+            <SocialCount>
+              <BranchLogo />
+              <TextSpan>{forkCount}</TextSpan>
+            </SocialCount>
+          )}
+        </SocialWrapper>
+      )}
```

Comment:
I find this to be a bit hard to read because of all the embedded logic. Is there a way we could extract this to be cleaner?

Reply:
Yea, that's possible.